### ISSUE TITLE
Hex.pm Community Package for upstream Elixir Dependents

### ISSUE THREAD

**Issue by cloudbring** [cmullaparthi/ibrowse#128]
Multiple of Elixir projects depend on `ibrowse` (who are in turn dependents for other projects in the Elixir ecosystem)
- [**HTTPotion**](https://github.com/myfreeweb/httpotion)
  - ibrowse hex package..interest? - myfreeweb/httpotion#56
- [**ExGrid**](https://github.com/chrismccord/phoenix_haml)
  - Publish as Hex package - bradleyd/exgrid#2
- [**ExVCR**](https://github.com/parroty/exvcr)

Please register an ibrowse Hex package so these downstream projects can use that definition rather than having to use both mix and hex to get ibrowse.

I'd be willing to setup ibrowse as a Hex [**community package**](https://groups.google.com/forum/#!topic/elixir-lang-talk/EapnMf_yKmQ)

**Comment by cloudbring** [comment]
Updated description and linked to myfreeweb/httpotion#56

**Comment by cmullaparthi** [comment]
Hi Emmanuel. It would be great if you could help setup ibrowse as a Hex community package. Let me know if you need anything from me.

**Comment by cloudbring** [comment]
Alright, I'll get started on it. I'm doing this project kinda nights and weekends so it might be a bit but, if you don't see any progress by this week, feel free to ping me or reply to this thread. Thankfully, Gmail flags everything from github not only as not spam but, highest priority in my inbox. Someone at Google, clearly knows I'm a programmer.

**Comment by cmullaparthi** [comment]
Knock knock.

**Comment by josevalim** [comment]
@cmullaparthi since Rebar 3 is getting more and more traction and it also uses Hex, would you publish it to Hex if we help with the setup? We want to rely less on community packages and help authors maintain their own packages more and more.

**Comment by zmoshansky** [comment]
@cmullaparthi Would you consider publishing directly to hex? 
Tutorial: https://hex.pm/docs/publish
PR: has a mix file already that should cover most of the work https://github.com/hexpm/community/pull/71

**Comment by josevalim** [comment]
Ha, we commented at the same time. :D

Note we have also just added notes for deploying with rebar3, which should be more at hand for Erlang developers: https://hex.pm/docs/rebar3_publish

**Comment by cmullaparthi** [comment]
Yes that would be great! Some help would go a long way. 

--
W: http://chandrusoft.wordpress.com

**Comment by cmullaparthi** [comment]
Okay thanks, will look into this. 

--
W: http://chandrusoft.wordpress.com

**Comment by cmullaparthi** [comment]
Hi guys,

I've tried doing this, but publishing the package fails with "Unknown server error". Any idea what is happening?

```
$ rebar3 hex publish
Publishing ibrowse 4.2
  Dependencies:
    
  Excluded dependencies (not part of the Hex package):
    
  Included files:
    src/Emakefile.src
    src/ibrowse.app.src
    src/ibrowse.erl
    src/ibrowse_app.erl
    src/ibrowse_http_client.erl
    src/ibrowse_lb.erl
    src/ibrowse_lib.erl
    src/ibrowse_socks5.erl
    src/ibrowse_sup.erl
    include/ibrowse.hrl
    priv/ibrowse.conf
    rebar.config
    README.md
    LICENSE
Before publishing, please read Hex CoC: https://hex.pm/docs/codeofconduct
Proceed? ("Y")> 
===> Unknown server error
```

**Comment by henrik** [comment]
@cmullaparthi Are you still getting that error? If you do, I suspect @ericmj might be the person to ask about "Unknown server error" with Hex.

**Comment by ericmj** [comment]
I cannot reproduce the "Unknown server error" but Hex only supports semver and `4.2` is not a valid version. It needs to be `4.2.0`.

**Comment by cmullaparthi** [comment]
@ericmj, I've tried with 4.2.1 and it now fails with a different error. No idea what to do.

```
$ rebar3 hex publish
Publishing ibrowse 4.2.1
  Dependencies:
    
  Excluded dependencies (not part of the Hex package):
    
  Included files:
    src/Emakefile.src
    src/ibrowse.app.src
    src/ibrowse.erl
    src/ibrowse_app.erl
    src/ibrowse_http_client.erl
    src/ibrowse_lb.erl
    src/ibrowse_lib.erl
    src/ibrowse_socks5.erl
    src/ibrowse_sup.erl
    include/ibrowse.hrl
    priv/ibrowse.conf
    rebar.config
    rebar.lock
    README.md
    LICENSE
Before publishing, please read Hex CoC: https://hex.pm/docs/codeofconduct
Proceed? ("Y")> 
===> Hex Error: account not authorized for this action
```

**Comment by cmullaparthi** [comment]
I've tried a few things to resolve this.

```
$ rebar3 hex user auth
Username: ([])> cmullaparthi
Password: 
Generating API key...
Generation of API key failed (401)
```

```
$ rebar3 hex user whoami
cmullaparthi
```

**Comment by cmullaparthi** [comment]
```
$ rebar3 hex user reset_password
Username or Email: ([])> cmullaparthi
===> Uncaught error in rebar_core. Run with DEBUG=1 to see stacktrace
===> When submitting a bug report, please include the output of `rebar3 report "your command"`
```

**Comment by ericmj** [comment]
@cmullaparthi If you find rebar bugs, please report them to rebar.

It seems like you weren't the owner of ibrowse. I have reset the package, can you try pushing again?

**Comment by cmullaparthi** [comment]
Thanks @ericmj 

```
===> Hex Error: Validation failed
	package: meta: links: expected type dict(string, string)
```

**Comment by ericmj** [comment]
The error is saying you have the wrong type for the value in the links field here https://github.com/cmullaparthi/ibrowse/blob/master/src/ibrowse.app.src#L10. It should be a list of 2-item tuples, see here https://hex.pm/docs/rebar3_publish.

**Comment by cmullaparthi** [comment]
Thank you, that fixed it.

```
===> Published ibrowse 4.2.2
```

**Comment by henrik** [comment]
@cmullaparthi :tada: :+1: Awesome, thanks for doing this!
